### PR TITLE
Support Aggregate push down for incremental scan

### PR DESCRIPTION
**Pull request by huaxingao**
Enable Aggregate push down for incremental scan. Here is the original [PR](https://github.com/apache/iceberg/pull/7636)

### CODE REVIEW

**Comment by huaxingao** [comment]
In this PR, I have basically extracted the code for building an `org.apache.iceberg.Scan` from `buildBatchScan()` and reused it in the `pushAggregation` method. The difference between the `org.apache.iceberg.Scan` in `pushAggregation` and `buildBatchScan` is that stats are needed in `pushAggregation`, which is why I added a `withStats` flag. This PR is exactly the same as the old one, except I pass in `expectedSchema` so I don't need to call `schemaWithMetadataColumns()` twice.

**Comment by huaxingao** [comment]
cc @szehon-ho Could you please take a look when you have a moment? Thanks a lot!

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestAggregatePushDown.java [hunk 1]
```diff
@@ -808,4 +813,78 @@ public void testInfinity() {
...
+            .option(SparkReadOptions.END_SNAPSHOT_ID, snapshotId3)
+            .load(tableName)
+            .agg(functions.min("data"), functions.max("data"), functions.count("data"));
+
+    String explain1 =
+        pushdownResult.queryExecution().explainString(ExplainMode.fromString("simple"));
+
+    assertThat(explain1).contains("LocalTableScan", "min(data)", "max(data)", "count(data)");
+
+    Dataset<Row> noPushdownResult =
```
[the remark this answers is not in the document]
Comment (amogh-jahagirdar) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestAggregatePushDown.java:844]: Also I don't think there's too much additional value in verifying the noPushdown case here. I think the test should just enable agg pushdown, and explicitly verify the expected results. But not super opinionated.

[the remark this answers is not in the document]
Comment (huaxingao) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestAggregatePushDown.java]: Thanks @amogh-jahagirdar for the review! I have changed the variable names and removed the noPushdown cases.

**Review by amogh-jahagirdar — COMMENTED** [review]
Thank you @huaxingao the change looks great to me. I'll wait in case @szehon-ho or others have comments before merging.

**Review by amogh-jahagirdar — APPROVED** [review]

**Review by szehon-ho — APPROVED** [review]
lgtm, thanks!

**Comment by szehon-ho** [comment]
Merged, thanks @huaxingao

**Comment by huaxingao** [comment]
Thanks @amogh-jahagirdar, @szehon-ho, and of course, @RussellSpitzer

**Comment by huaxingao** [comment]
I will have a follow-up PR to port the changes to Spark 3.4. @szehon-ho

**Pull request merged**